drivers/pinctrl/intel/pinctrl-intel.c

```diff
@@ -858,6 +864,15 @@ static void intel_gpio_irq_enable(struct irq_data *d)
 		value |= BIT(gpp_offset);
 		writel(value, community->regs + community->ie_offset + gpp * 4);
 		raw_spin_unlock_irqrestore(&pctrl->lock, flags);
+
+		community->features |= PINCTRL_FEATURE_INT_CHECK_TIMER;
+	}
+
+	if (pctrl->irq_enabled == false) {
+		pctrl->irq_enabled = true;
+		// Trigger timer only on APL
```

Comment:
Lets do this on all platforms for now, no need to filter for APL; I don't think there are any users of intel-gpio GPIO IRQ interrupts that are not broken in this way.

Reply:
actually lets be a bit safer - can you make it read the PADCFG register here and only trigger for PADCFG0_RXEVCFG_LEVEL interrupts

Reply:
If I move the mod_timer to the irq_handler, then I think we don't need mod_timer in irq_enable.